pkg/webhook/client.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// IsPreprovisionCSIVolume checks whether the volume is a pre-provisioned volume for the desired csiDriver.
+func (si *SidecarInjector) IsPreprovisionCSIVolume(csiDriver string, pvc *corev1.PersistentVolumeClaim) (bool, error) {
+	_, ok, err := si.GetPreprovisionCSIVolume(csiDriver, pvc)
+
+	return ok, err
+}
+
+// GetPreprovisionCSIVolume gets the pre-provisioned persistentVolume when backed by the desired csiDriver.
+func (si *SidecarInjector) GetPreprovisionCSIVolume(csiDriver string, pvc *corev1.PersistentVolumeClaim) (corev1.PersistentVolume, bool, error) {
+	var pv corev1.PersistentVolume
+	if csiDriver == "" {
+		return pv, false, errors.New("csiDriver is empty, cannot verify storage type")
+	}
+
+	if pvc == nil {
+		return pv, false, errors.New("pvc is nil, cannot get pv")
+	}
+
+	if pvc.Spec.VolumeName == "" {
```

Comment:
why is returning nil?

Reply:
@saikat-royc we are returning a nil error because the volume is valid. A pvc can have this field missing when its requesting a dynamically provisioned volume and it is not yet bound.

Reply:
@saikat-royc will leave up to you to resolve this comment.

Reply:
Ack. Please put a comment in the code for this and resolve this conversation